QuizApp/QuizApp/Network/CheckNetworkClient.swift

```
@@ -0,0 +1,40 @@
+import Foundation
+
+protocol CheckNetworkClientProtocol {
+
+    func check() async throws
+
+}
+
+class CheckNetworkClient: NetworkClient, CheckNetworkClientProtocol {
+
+    private let baseUrl: String
+
+    init(baseUrl: String) {
+        self.baseUrl = baseUrl
+    }
+
+    func check() async throws {
+        guard let url = URL(string: "\(baseUrl)/v1/check") else {
+            throw RequestError.invalidURLError
+        }
+
+        guard let accessToken = SecureStorage().fetchAccessToken() else {
+            throw CheckNetworkError.noAccessTokenError
```

I do not think we need to throw an error if the token is not present. The server will return 401 and you can check the response error and throw error there.